File: app/src/main/java/com/juawapps/newsspread/di/AppModuleImageLoader.java
package com.juawapps.newsspread.di;

import com.juawapps.newsspread.image.ImageLoaderWrapper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Image loader module
 */
@Module
public class AppModuleImageLoader {
    @Singleton
    @Provides
    ImageLoaderWrapper provideImageLoaderWrapper() {
        return new ImageLoaderWrapper();
    }
}

File: app/src/main/java/com/juawapps/newsspread/di/AppModuleImageLoader.kt
package com.juawapps.newsspread.di

import com.juawapps.newsspread.image.ImageLoaderWrapper

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Image loader module
 */
@Module
class AppModuleImageLoader {

    @Singleton
    @Provides
    internal fun provideImageLoaderWrapper(): ImageLoaderWrapper {
        return ImageLoaderWrapper()
    }
}
